File: project/backend_server/server/src/main/java/com/sustech/groupup/utils/Response.java
package com.sustech.groupup.utils;

import lombok.Data;

@Data
public class Response {
    private int code;
    private String msg;
    private Object data;

    public Response(ResponseCodeType codeType, String msg, Object data) {
        this.code = codeType.getCode();
        this.msg = msg;
        this.data = data;
    }

    //success
    public static Response getSuccess(String msg, Object data) {
        return new Response(ResponseCodeType.SUCCESS, msg, data);
    }

    public static Response getSuccess(Object data) {
        System.out.println(data);
        System.out.println(new Response(ResponseCodeType.SUCCESS, "success", data));
        return new Response(ResponseCodeType.SUCCESS, "", data);
    }

    public static Response getSuccess() {
        return new Response(ResponseCodeType.SUCCESS, "", null);
    }

    //not modified
    public static Response getNotModified(String msg) {
        return new Response(ResponseCodeType.NOT_MODIFIED, msg, null);
    }

    public static Response getNotModified() {
        return new Response(ResponseCodeType.NOT_MODIFIED, "", null);
    }

    //bad request
    public static Response getBadRequest(String msg) {
        return new Response(ResponseCodeType.BAD_REQUEST, msg, null);
    }

    //unauthorized
    public static Response getUnauthorized(String msg) {
        return new Response(ResponseCodeType.UNAUTHORIZED, msg, null);
    }

    //not found
    public static Response getNotFound(String msg) {
        return new Response(ResponseCodeType.NO_FOUND, msg, null);
    }

    public static Response getInternalError(String msg) {
        return new Response(ResponseCodeType.INTERNAL_ERROR, msg, null);
    }
}

File: project/backend_server/recommendation/common/Response.py
from enum import Enum
import json

class ResponseCodeType(Enum):
    SUCCESS = 200
    NOT_MODIFIED = 304
    BAD_REQUEST = 400
    UNAUTHORIZED = 401
    NO_FOUND = 404
    INTERNAL_ERROR = 500

class Response:
    def __init__(self, code_type, msg, data):
        self.code = code_type.value
        self.msg = msg
        self.data = data

    @staticmethod
    def get_success(msg="", data=None):
        return Response(ResponseCodeType.SUCCESS, msg, data)

    @staticmethod
    def get_not_modified(msg=""):
        return Response(ResponseCodeType.NOT_MODIFIED, msg, None)

    @staticmethod
    def get_bad_request(msg):
        return Response(ResponseCodeType.BAD_REQUEST, msg, None)

    @staticmethod
    def get_unauthorized(msg):
        return Response(ResponseCodeType.UNAUTHORIZED, msg, None)

    @staticmethod
    def get_not_found(msg):
        return Response(ResponseCodeType.NO_FOUND, msg, None)

    @staticmethod
    def get_internal_error(msg):
        return Response(ResponseCodeType.INTERNAL_ERROR, msg, None)

    def to_dict(self):
        return {
            'code': self.code,
            'msg': self.msg,
            'data': self.data
        }

    def __str__(self):
        return json.dumps(self.to_dict())